### PR TITLE
Implements Expert Review Section

### PR DESCRIPTION
**Pull request by saengowp**
- Add `/expert/:id`
- The entire process incl. previous PR took 3.5 hours
image

### CODE REVIEW

**Comment by saenyakorn** [comment]
Demo pls

File: apps/doji-frontend/src/components/Review/RatingEntry.tsx [hunk 1]
```diff
@@ -11,7 +11,8 @@ function ReviewEntry(props: { data: IPublicSessionReviewDTO }) {
     <Stack py="1em" spacing="0.75em">
       <Stack direction="row" alignItems="center" justifyContent="space-between">
         <Stack direction="row" spacing="0.5em" alignItems="center">
-          <Avatar src="https://mui.com/static/images/avatar/1.jpg" sx={{ width: 24, height: 24 }} />
+          //TODO Avatar
```
Comment (saenyakorn) [inline apps/doji-frontend/src/components/Review/RatingEntry.tsx:14]: To comment in React component. Do this

```jsx
return (
  <div>
  {/* This is a comment */}
  </div>
)
```

File: apps/doji-backend/src/session/session.dto.ts [hunk 2]
```diff
@@ -91,3 +95,25 @@ export class SessionDTO implements ISession {
...
+  @ApiProperty()
+  id: number
+  @ApiProperty()
+  rating: number
+  @ApiProperty()
+  authorName: string
+  @ApiProperty()
+  content: string
+  @ApiProperty()
+  createdAt: Date
```
Comment (saenyakorn) [inline apps/doji-backend/src/session/session.dto.ts:118]: spacing

File: apps/doji-frontend/src/pages/expert/[id]/index.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { useRouter } from 'next/router'
+
+import { useQuery } from 'react-query'
+
+const ExpertPage: ExtendedNextPage = () => {
+  //TODO This Entire Page
+  const router = useRouter()
+  const { id } = router.query
+  const { isError, error, isLoading, data } = useQuery<IExpertInfoDTO>(['/expert/info', id], () =>
+    httpClient.get('/expert/info/' + encodeURIComponent(id as string)).then((res) => res.data),
```
Comment (saenyakorn) [inline apps/doji-frontend/src/pages/expert/[id]/index.tsx:16]: wonder why we need `encodeURIComponent` since `id` is already alphabet

**Review by saenyakorn — APPROVED** [review]
Other than that LGTM. I guess

File: apps/doji-backend/src/session/session.dto.ts (hunk 2 above)
Comment (saenyakorn) [reply apps/doji-backend/src/session/session.dto.ts:118]: I mean 
```suggestion
  @ApiProperty()
  id: number
  
  @ApiProperty()
  reviews: IPublicSessionReviewDTO[]
  
  @ApiProperty()
  reviewStat: ReviewStatDTO
}

export class PublicSessionReviewDTO implements IPublicSessionReviewDTO {
  @ApiProperty()
  id: number
  
  @ApiProperty()
  rating: number
  
  @ApiProperty()
  authorName: string
  
  @ApiProperty()
  content: string
  
  @ApiProperty()
  createdAt: Date
```

Comment (saenyakorn) [reply apps/doji-backend/src/session/session.dto.ts:118]: Oh I forgot. name the DTO end with `Response` or `Request` like 

`PublicSessionReviewResponseDTO `

Comment (saenyakorn) [reply apps/doji-backend/src/session/session.dto.ts:118]: And should we use `class-validator` on this?

File: apps/doji-frontend/src/components/Review/RatingEntry.tsx (hunk 1 above)
Comment (saengowp) [reply apps/doji-frontend/src/components/Review/RatingEntry.tsx:14]: ok

File: apps/doji-frontend/src/pages/expert/[id]/index.tsx (hunk 3 above)
Comment (saengowp) [reply apps/doji-frontend/src/pages/expert/[id]/index.tsx:16]: ok

File: apps/doji-backend/src/session/session.dto.ts (hunk 2 above)
Comment (saengowp) [reply apps/doji-backend/src/session/session.dto.ts:118]: - Ok
- No need. It is output only

**Comment by saengowp** [comment]
I spent 4 hour doing a failed git surgery to get this demo up, pls review

Demo: https://dev.choco.saenyakorn.dev/expert/pipat

File: apps/doji-backend/src/session/session.dto.ts [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
   IGetServiceByNameAndExpertUsernameDTO,
-  IPublicSessionReviewDTO,
+  IPublicSessionReviewDTO as IPublicSessionReviewResponseDTO,
   IReviewStat,
   IScheduleSessionDTO,
   IService,
   IServiceInformationDTO,
   ISession,
-  ISessionInformationDTO,
+  ISessionInformationDTO as ISessionInformationResponseDTO,
```
Comment (saenyakorn) [inline apps/doji-backend/src/session/session.dto.ts:12]: name alias 😥

**Review by saenyakorn — APPROVED** [review]
The demo is great. Other than that LGTM

**Comment by saenyakorn** [comment]
ปล. แก้เรื่อง name แล้ว merge ได้เลย

File: apps/doji-backend/src/session/session.dto.ts (hunk 4 above)
Comment (saengowp) [reply apps/doji-backend/src/session/session.dto.ts:12]: Oops. My bad. Auto-rename tools is weird.

**Pull request merged**